sled-agent/zone-images/src/source_resolver.rs

```diff
@@ -64,20 +85,47 @@ impl ZoneImageSourceResolver {
 
 #[derive(Debug)]
 struct ResolverInner {
+    #[expect(unused)]
+    log: slog::Logger,
     image_directory_override: Option<Utf8PathBuf>,
+    // Store all collected information for mupdate overrides -- we're going to
+    // need to report this via inventory.
+    //
+    // This isn't actually used yet.
+    #[expect(unused)]
+    mupdate_overrides: AllMupdateOverrides,
 }
 
 impl ResolverInner {
-    fn new() -> Self {
-        Self { image_directory_override: None }
+    fn new(
+        log: &slog::Logger,
+        zpools: &ZoneImageZpools<'_>,
+        boot_zpool: &ZpoolName,
+    ) -> Self {
+        let log = log.new(o!("component" => "ZoneImageSourceResolver"));
+
+        let mupdate_overrides =
+            AllMupdateOverrides::read_all(&log, zpools, boot_zpool);
+
+        // Log the results.
+        mupdate_overrides.log_results(&log);
```

Comment:
Tiny nit - why not do this implicitly inside `read_all`?

Reply:
Seems reasonable, fixed.